### DISCUSSION TITLE
would local replica and local filer speed up to transfers?

### DISCUSSION THREAD

**Discussion by eleaner** [seaweedfs/seaweedfs#3471] [General]
I work a lot with photos (so rather big files, often exceeding 1GB)
I am a bit paranoid about data loss and accessibility, so I am running a home lab cluster of 16 machines as a distributed storage cluster on software competing with seaweed
sadly only over 1G network, so the transfers are OK but not great.

my main working machine is a mac
thinking of seaweed, I could potentially stick a few HDD in it and have an additional replica of the data locally
I could also run an additional local filer with a distributed data store shared with the rest of the cluster.

Would such setup speed up local access to seaweed data from my mac (or any machine for that matter)?

**Comment by chrislusf** [comment]
Try to run "weed mount" on the Mac with a big cache size.

**Reply by eleaner** [reply]
as I understand, a big cache size would speed up writes but not reads unless the file is heavily in use.
am I right?

my idea was that the local replica could be read with a local speed rather than going through the 1G network

**Reply by chrislusf** [reply]
ok. this should help https://github.com/seaweedfs/seaweedfs/wiki/Cloud-Drive-Architecture

**Reply by eleaner** [reply]
I read it and I think you misunderstood my question. 

I am thinking SOHO, not enterprise storage. I don't even look at the cloud (yet)
my cluster sits in the next room connected by a 1G network. 
I am not a network engineer, but I can probably get between 40-100MB/s
that's kind of slow if I read 1,5G of the file

so "local" means HDD(s) physically inside of my mac pro - potentially getting much faster speeds than dragging the data via the network from the other room.

I assume I could set up a "warm storage cluster" inside of my mac(?)
but I don't want to store multiple copies of data there

**Reply by chrislusf** [reply]
ok. seems mount is easier. You can warm the cache.

**Reply by eleaner** [reply]
do you mean like massive few TB SSD dedicated as a cache for weed mount?

**Reply by eleaner** [reply]
@chrislusf 
how would I specify, say 2TB of SSD, as a cache for my mounted seaweedfs?

**Reply by chrislusf** [reply]
see `weed mount -h`

**Reply by eleaner** [reply]
stupid me

  -cacheCapacityMB int
        local file chunk cache capacity in MB
  -cacheDir string
        local cache directory for file chunks and meta data (default "/tmp")
